src/fw/drivers/sf32lb52/temperature.c

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2025 SiFli Technologies(Nanjing) Co., Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "drivers/temperature.h"
+
+#include "bf0_hal.h"
+#include "board/board.h"
+#include "console/prompt.h"
+#include "drivers/gpio.h"
+#include "drivers/periph_config.h"
+#include "drivers/voltage_monitor.h"
+#include "kernel/util/sleep.h"
+#include "mfg/mfg_info.h"
+#include "services/common/regular_timer.h"
+#include "system/logging.h"
+#include "system/passert.h"
+
+static TSEN_HandleTypeDef TSEN_Handle = {.Instance = hwp_tsen};
+
+void temperature_init(void) {
+  if (HAL_TSEN_Init(&TSEN_Handle) != HAL_OK) {
+    PBL_ASSERTN(0);
+  }
+}
+
+static void prv_tsen_enable(TSEN_HandleTypeDef *htsen) {
+  htsen->Instance->TSEN_CTRL_REG &= ~TSEN_TSEN_CTRL_REG_ANAU_TSEN_RSTB;
+  htsen->Instance->TSEN_CTRL_REG |=
+      TSEN_TSEN_CTRL_REG_ANAU_TSEN_EN | TSEN_TSEN_CTRL_REG_ANAU_TSEN_PU;
+  htsen->Instance->TSEN_CTRL_REG |= TSEN_TSEN_CTRL_REG_ANAU_TSEN_RSTB;
+  HAL_Delay_us(20);
+  htsen->Instance->TSEN_CTRL_REG |= TSEN_TSEN_CTRL_REG_ANAU_TSEN_RUN;
+  htsen->State = HAL_TSEN_STATE_ENABLED;
```

Comment:
looks like you're using a native driver, so you can likely just use `hwp_tsen` everywhere and save useless State